cfg/posix.cfg

```diff
@@ -2928,6 +2928,36 @@ The function 'mktemp' is considered to be dangerous due to race conditions and s
     </arg>
     <warn severity="style" reason="Obsolescent" alternatives="pthread_attr_getstack"/>
   </function>
+  <!-- int pthread_attr_setstacksize(pthread_attr_t *attr, size_t stacksize); -->
+  <!-- int pthread_attr_setguardsize(pthread_attr_t *attr, size_t guardsize); -->
+  <function name="pthread_attr_setstacksize,pthread_attr_setguardsize">
+    <noreturn>false</noreturn>
+    <returnValue type="int"/>
+    <leak-ignore/>
+    <arg nr="1">
+      <not-null/>
+      <not-uninit/>
+    </arg>
+    <arg nr="2">
+      <not-uninit/>
+      <not-bool/>
```

Comment:
you could also add `<valid>1:</valid>` maybe

Reply:
Thought about that too, but there could be false positives.
For `pthread_attr_setstacksize()`:
According to https://www.mkssoftware.com/docs/man3/pthread_attr_setstacksize.3.asp:
"On Windows, setting the thread stack size to 0 causes the thread's stack size to be inherited from the parent thread. This is the default setting."
And for `pthread_attr_setguardsize()`:
According to https://linux.die.net/man/3/pthread_attr_setguardsize:
"If guardsize is 0, then new threads created with attr will not have a guard area."

For the guardsize 0 is perfectly valid.
For the stacksize there could be special (maybe a bit strange) cases (Windows) where 0 is valid / intended. Not sure if `<valid>1:</valid>` is ok then.

Reply:
ok. how about preventing negative values?

Reply:
That would be ok. I was not sure if it is necessary or useful to add it for arguments with type `size_t`.

Reply:
It will be a good idea, because the function declaration is likely not seen.

Maybe it would be good to add the type info for arguments in the configuration somehow. If size_t is a 32-bit integer it would be nice to warn about 64-bit arguments.